src/overlays/actors/ovl_En_Famos/z_en_famos.c

```diff
@@ -28,133 +66,792 @@ const ActorInit En_Famos_InitVars = {
+ */
+void EnFamos_SetupAttackDebris(EnFamos* this) {
+    EnFamosRock* rock;
+    f32 randFloat;
+    s16 randOffset;
+    s16 randVelDirection;
+    s32 i;
+
+    this->debrisTimer = 40;
+    rock = &this->rocks[0];
+    for (i = 0; i < ARRAY_COUNT(this->rocks); i++, rock++) {
+        randVelDirection = Rand_Next() >> 0x10;
+        randOffset = Rand_S16Offset(0x1800, 0x2800);
+        randFloat = Rand_ZeroFloat(5.0f) + 5.0f;
+        rock->velocity.x = randFloat * Math_CosS(randOffset) * Math_SinS(randVelDirection);
+        rock->velocity.y = Math_SinS(randOffset) * randFloat + 3.0f;
+        rock->velocity.z = randFloat * Math_CosS(randOffset) * Math_CosS(randVelDirection);
+        rock->rotation.x = Rand_Next() >> 0x10;
+        rock->rotation.y = Rand_Next() >> 0x10;
+        rock->rotation.z = Rand_Next() >> 0x10;
+        rock->pos.x = (Math_SinS(randVelDirection) * 20.0f) + this->actor.world.pos.x;
+        rock->pos.y = this->actor.floorHeight;
+        rock->pos.z = (Math_CosS(randVelDirection) * 20.0f) + this->actor.world.pos.z;
+        rock->scale = Rand_ZeroFloat(0.0015f) + (2.0f * 0.001f);
+    }
+}
+
+/**
+ * Summary: Sets 20 rocks to draw from an explosion on death.
+ */
+void EnFamos_SetupDeathDebris(EnFamos* this) {
+    f32 randFloat;
+    s16 randSmaller;
+    s16 randVelDirection;
+    EnFamosRock* rock;
+    s32 i;
+
+    this->debrisTimer = 40;
+    rock = &this->rocks[0];
+    for (i = 0; i < ARRAY_COUNT(this->rocks); i++, rock++) {
+        randVelDirection = Rand_Next() >> 0x10;
+        randSmaller = (u32)Rand_Next() >> 0x12;
+        randFloat = Rand_ZeroFloat(6.0f) + 7.0f;
+        rock->velocity.x = randFloat * Math_CosS(randSmaller) * Math_SinS(randVelDirection);
+        rock->velocity.y = Math_SinS(randSmaller) * randFloat + 4.5f;
+        rock->velocity.z = randFloat * Math_CosS(randSmaller) * Math_CosS(randVelDirection);
+        rock->rotation.x = Rand_Next() >> 0x10;
+        rock->rotation.y = Rand_Next() >> 0x10;
+        rock->rotation.z = Rand_Next() >> 0x10;
+        rock->pos.x = Math_SinS(randVelDirection) * 20.0f + this->actor.world.pos.x;
+        rock->pos.y = randPlusMinusPoint5Scaled(60.0f) + (this->actor.world.pos.y + 40.0f);
+        rock->pos.z = Math_CosS(randVelDirection) * 20.0f + this->actor.world.pos.z;
+        rock->scale = Rand_ZeroFloat(0.002f) + (2.5f * 0.001f);
+    }
+}
+
+s32 EnFamos_IsPlayerSeen(EnFamos* this, GlobalContext* globalCtx) {
+    if (Player_GetMask(globalCtx) != PLAYER_MASK_STONE &&
+        Actor_XZDistanceToPoint(&GET_PLAYER(globalCtx)->actor, &this->calmPos) < this->agroDistance &&
+        (Actor_IsFacingPlayer(&this->actor, 0x5000))) {
+        return true;
+    } else {
+        return false;
+    }
+}
+
+void EnFamos_UpdateBobbingHeight(EnFamos* this) {
+    if (this->hoverClk == 0) {
+        this->hoverClk = 30;
+    }
+
+    this->hoverClk--;
+    this->actor.world.pos.y = (Math_SinS(this->hoverClk * 2184) * 10.0f) + this->baseHeight;
```

Comment:
```suggestion
    this->actor.world.pos.y = (Math_SinS(this->hoverClk * 0x888) * 10.0f) + this->baseHeight;
```
although it's probably something sensible in degrees anyway.